paddle/fluid/operators/cub_reduce.h

```diff
@@ -293,7 +294,10 @@ void TensorReduce(const framework::Tensor& x, framework::Tensor* y,
   }
   auto x_data = x.data<Tx>();
   auto y_data = y->mutable_data<Ty>(x.place());
-  if (reduce_num == 1) return;
+  if (reduce_num == 1) {
+    framework::TensorCopy(x, y->place(), y);
```

Comment:
`TensorCopy` will resize y to x.shape. So we should restore the shape of y after `TensorCopy`.

Reply:
Done. Thanks!